libc-test/build.rs

```diff
@@ -281,6 +281,9 @@ fn main() {
         match ty {
             "sockaddr_nl" => musl,
 
+            // user-defined struct to patch struct utmpx on Linux
+            "__timeval" if linux => true,
```

Comment:
The declaration of `struct utmpx` on Linux:
```
struct utmp {
...
...

    /* The ut_session and ut_tv fields must be the same size when
       compiled 32- and 64-bit.  This allows data files and shared
       memory to be shared between 32- and 64-bit applications. */
#if __WORDSIZE == 64 && defined __WORDSIZE_COMPAT32
    int32_t ut_session;           /* Session ID (getsid(2)),
                                     used for windowing */
    struct {
        int32_t tv_sec;           /* Seconds */
        int32_t tv_usec;          /* Microseconds */
    } ut_tv;                      /* Time entry was made */
#else
     long   ut_session;           /* Session ID */
     struct timeval ut_tv;        /* Time entry was made */
#endif

...
};
```
As shown above, the type of field `ut_tv` can be an anonymous struct, so I have to define a `struct __timeval` which is nonexistent in glibc to handle this case.